### PR TITLE
Move cherry-pick automation to ADO

### PR DESCRIPTION
**Pull request by trungams**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

- Move cherry-picking automation from Github Actions to ADO

###### Change Log  

- Remove cherry-pick github workflow files
- Add pipeline definition for cherry-picking

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Example PR that bumps a package version: https://github.com/microsoft/CBL-Mariner/pull/6384
- Example PR created by cherry-pick automation: https://github.com/microsoft/CBL-Mariner/pull/6386

### CODE REVIEW

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:39]: Is it possible to make this pipeline live in our ADO repositories entirely? This pipeline will require additional permissions and secrets access and there's no need to risk exposing it to malicious modifications from forked repos, which may slip through the cracks.

Comment (trungams) [reply .pipelines/cherrypick/CherryPick.yml:39]: I looked into that, but unfortunately resources trigger doesn't work for Github repos as of now: https://learn.microsoft.com/en-us/azure/devops/pipelines/repos/multi-repo-checkout?view=azure-devops#triggers. 

> This pipeline will require additional permissions and secrets access

If I remember correctly, the default permissions won't allow forks to start this pipeline.

Comment (PawelWMS) [reply .pipelines/cherrypick/CherryPick.yml:39]: Nevermind, looks like [this is not supported](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/resources-repositories-repository?view=azure-pipelines#remarks). In that case, could you add a line in `.github/CODEOWNERS` to make any changes to this file require an approval from `@microsoft/cbl-mariner-admins`?

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:8]: Adding a comment, so we don't forget to remove this line before merging.

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:4]: Maybe explicitly mention that this is aimed at fast-track commits.

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:26]: Do we need this specific pool? Or would a generic `linux` be OK?

If we do need this specific agent, please use the `Agent pools (DEV)` group and the `$(DEV_AMD64_Managed)` pool inside it. This way, if the name ever changes, we won't break this pipeline.

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:33]: Can we move `chekout` inside `CherryPick.yml`? This way it'll have control over the folder where the repository is cloned and we can drop the `variables` here and give more flexibility to the template.

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [reply .pipelines/cherrypick/CherryPick.yml:39]: And yes, we can (even must) disable fork builds, but I want to minimize a chance of a malicious PR getting approved and merged.

File: .pipelines/cherrypick/CherryPick.yml
Comment (trungams) [reply .pipelines/cherrypick/CherryPick.yml:26]: Oh I was not aware of this. I was using the same agent pool used by autopatching/autoupgrade tools. I'm not sure if generic linux would work, since the script that installs dependencies relies on the OS being ubuntu. That can change in the future though, so we can revisit this later.

File: .pipelines/cherrypick/CherryPick.yml
Comment (trungams) [reply .pipelines/cherrypick/CherryPick.yml:33]: sorry, I'm a bit confused, do you mean moving the checkout step to `CherryPick.yml` of CBL-Mariner-Pipelines?

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [reply .pipelines/cherrypick/CherryPick.yml:26]: No problem, it's a fairly new variable group. And if you're saying that Ubuntu is what we need, then use `$(DEV_AMD64_Ubuntu_Managed)`.

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [reply .pipelines/cherrypick/CherryPick.yml:33]: Ah, both files have the same name - I didn't see that. To answer your question: both. I meant moving the `checkout` step to `CherryPick.yml` in `CBL-Mariner-Pipelines`.:)

File: .pipelines/cherrypick/CherryPick.yml
Comment (trungams) [reply .pipelines/cherrypick/CherryPick.yml]: I updated `.github/CODEOWNERS`. Since we don't have a `pipelines/` folder (without the dot), I updated that entry to `.pipelines/`

File: .github/CODEOWNERS
Comment (PawelWMS) [inline .github/CODEOWNERS:8]: Thanks for catching that!

File: .pipelines/cherrypick/CherryPick.yml
Comment (PawelWMS) [inline .pipelines/cherrypick/CherryPick.yml:26]: No need for `agentPool`:
```suggestion

resources:
  repositories:
    - repository: CBL-Mariner-Pipelines
      type: git
      name: mariner/CBL-Mariner-Pipelines

jobs:
  - job: 'CherryPick'
    pool:
      type: linux
      isCustom: true
      name: "$(DEV_AMD64_Ubuntu_Managed)"
```

**Review by PawelWMS — APPROVED** [review]
One last comment but looks good.

**Pull request merged**
